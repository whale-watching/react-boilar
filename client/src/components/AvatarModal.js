import React from 'react';
import Modal from 'react-modal';
import axios from 'axios';
import { socketEmit } from '../helpers/socketEvents';

class AvatarModal extends React.Component {

  state = {
    error: null
  }

  previewAvatar = (input) => {
    if (input.files && input.files[0]) {
      const reader = new FileReader();

      reader.onload = function (e) {
        const preview = document.getElementById('preview');
        preview.src = e.target.result;
      }

      reader.readAsDataURL(input.files[0]);
    }
  }

  uploadAvatar = (e) => {
    e.preventDefault();

    const data = new FormData();
    const inputfile = document.getElementById('inputfile');

    data.append('id', this.props.user.id);
    data.append('avatar', inputfile.files[0]);

    axios.post('/api/avatar', data, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
    .then((res) => {
      socketEmit.getAvatar();
      this.props.onRequestClose();
    })
    .catch((err) => {
      this.setState({ error: 'Only .jpg, .png and .gif, < 5MB' });
    });
  }

  render() {
    return (
      <Modal
        className="avatar-modal"
        isOpen={this.props.isOpen}
        onRequestClose={this.props.onRequestClose}
      >
        <form onSubmit={(e) => this.uploadAvatar(e)}>
          <h3>Set Profile</h3>
          <p className="error">{this.state.error}</p>
          <img id="preview" src={this.props.user.avatar} />
          <input id="inputfile" name="avatar" type="file" onChange={() => this.previewAvatar(document.getElementById('inputfile'))}/>
          <button type="submit" className="button-text">Set</button>
        </form>
      </Modal>
    )
  }
}

export default AvatarModal;