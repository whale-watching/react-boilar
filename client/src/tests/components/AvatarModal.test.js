import React from 'react';
import { shallow } from 'enzyme';
import AvatarModal from '../../components/AvatarModal';
import avatar from '../data/avatar';

describe('<AvatarModal />', () => {
  it('renders AvatarModal correctly', () => {
    const wrapper = shallow(<AvatarModal isOpen={avatar.isOpen} onRequestClose={avatar.onRequestClose} user={avatar.user} />);

    expect(wrapper).toMatchSnapshot();
  });
});