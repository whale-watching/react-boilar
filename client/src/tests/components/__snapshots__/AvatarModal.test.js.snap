// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarModal /> renders AvatarModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="avatar-modal"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Set Profile
    </h3>
    <p
      className="error"
    />
    <img
      alt="avatar"
      id="preview"
      src="/img/avatars/1.png"
    />
    <input
      id="inputfile"
      name="avatar"
      onChange={[Function]}
      type="file"
    />
    <button
      className="button-text"
      type="submit"
    >
      Set
    </button>
  </form>
</Modal>
`;
